### PR TITLE
migrated stm32f3 based project to stm32f103rc-based system

### PR DESCRIPTION
**Pull request by francisfu-embedded**

### CODE REVIEW

**Comment by a2k-hanlon** [comment]
Hi Francis, thank you for your work here! I will look it over as soon as I can, but in the meantime, would you please restore the components/bms/bms_master/.gitignore file and remove the ignored files from the PR?

**Comment by francisfu-embedded** [comment]
Hi Andrew. I just recovered the .gitignore file. Just wondering what do you mean by "remove the ignored files from the PR"? Do you mean to remove the object files in the debug folder?

**Comment by a2k-hanlon** [comment]
Hi Francis, thank you for addressing that, but you have changed the wrong .gitignore file. We have more than 1 .gitignore file in this repository. You modified the .gitignore at the root of the repository rather than [components/bms/bms_master/.gitignore](https://github.com/UBC-Solar/firmware_v3/blob/master/components/bms/bms_master/.gitignore). Please revert the changes from your last commit and put back components/bms/bms_master/.gitignore.

Also, I see that what you put in that root .gitignore file is actually a bunch of HTML - I'm not sure what happened but please look over what you plan to commit before committing it.

To answer your question about removing the ignored files from the PR, I think this will make more sense once you see the bms_master folder .gitignore, but yes you are correct. The `Debug` folder within `components/bms/bms_master` needs to be removed. This folder was set to be ignored by git, but since a .gitignore file got deleted, you were able to inadvertently commit it with your migration changes.

**Comment by francisfu-embedded** [comment]
THanks Andrew. Sorry that I put the wrong .gitignore file. This one should work

File: components/bms/bms_master/project.yml [hunk 1]
```diff
@@ -1,108 +0,0 @@
----
```
Comment (a2k-hanlon) [inline components/bms/bms_master/project.yml]: Please restore this file too

File: components/bms/bms_master/test/test_selftest.c [hunk 2]
```diff
@@ -1,100 +0,0 @@
-#ifdef TEST
```
Comment (a2k-hanlon) [inline components/bms/bms_master/test/test_selftest.c]: Please restore this file too

**Comment by a2k-hanlon** [comment]
No worries, thank you Francis! Thank you for adding back those 2 unit test files. I have now checked over your reconfiguration and it all looks good. Nice work! Thanks for your patience with my review.

There is just one more file I'd like you to make a change for - `components/bms/bms_master/.mxproject`. This one was around before your changes here but I noticed that it contains computer-specific absolute filepaths, and it's a CubeMX-generated file that we don't need in version control. So, please delete this file from the repo and add `.mxproject` to the bms_master `.gitignore`.

I am approving your PR now, so just feel free to merge it to master once you've made that change. Thanks again!

**Review by a2k-hanlon — DISMISSED** [review]

**Comment by francisfu-embedded** [comment]
oh it seems like I'm blocked because of the changes I've made. Can you approve again?

**Review by a2k-hanlon — APPROVED** [review]

**Comment by francisfu-embedded** [comment]
merged

**Pull request merged**